File: Python/2021/Day2/part1.py
file = "input.txt"

inputs = []
with open(file, "r") as open_file:
    while (line := open_file.readline()):
        inputs.append(
            line.strip().split()
        )

horizontal = 0
depth = 0

for input in inputs:
    if input[0] == "forward":
        horizontal += int(input[1])

    elif input[0] == "down":
        depth += int(input[1])

    elif input[0] == "up":
        depth -= int(input[1])

print(horizontal*depth)

File: Java/2021/Day02/part1.java
import java.io.File;
import java.io.FileNotFoundException;
import java.util.*;

class Part1 {
  public static void main(String[] args) throws FileNotFoundException {
    ArrayList<String[]> data = readFile("input.txt");
    int horizontal = 0;
    int depth = 0;

    for (String[] action : data) {
      String command = action[0];
      int steps = Integer.parseInt(action[1]);

      switch (command) {
        case "forward":
          horizontal += steps;
          break;

        case "down":
          depth += steps;
          break;

        case "up":
        depth -= steps;
          break;
      }
    }

    System.out.println(horizontal*depth);
  }

  private static ArrayList<String[]> readFile(String path) throws FileNotFoundException {
    File myFile = new File(path);
    Scanner fileScanner = new Scanner(myFile);

    ArrayList<String[]> data = new ArrayList<String[]>();

    while (fileScanner.hasNextLine()) {
      String[] line = fileScanner.nextLine().trim().split(" ");

      data.add(line);
    }

    return data;
  }
}
